proxy/src/compute.rs

```diff
@@ -219,6 +218,16 @@ impl ConnCfg {
 
         Ok(connection)
     }
+
+    /// Connect to a corresponding compute node.
+    async fn connect(&self) -> Result<PostgresConnection, ConnectionError> {
+        self.do_connect()
+            .inspect_err(|err| {
+                // Immediately log the error we have at our disposal.
+                error!("couldn't connect to compute node: {err}");
```

Comment:
Not sure this belongs to this PR, but I think it would be very useful to have there endpoint name and/or timeline id + unique session_id (seems like it is in `ConsoleReqExtra`, but not sure it is logged here).

Reply:
Unique session id should be there. It's in the top-level span. Not sure about everything else: it's greppable by session_id, so no point in moving it here.